actors/builtin/storage_power/storage_power_actor_state.go

```diff
@@ -235,6 +256,54 @@ func (st *StoragePowerActorState) _getPledgeSlashForConsensusFault(currPledge ab
 	}
 }
 
+func asKeyer(a addr.Address) adt.Keyer {
+	return addrKeyWrapper{a}
+}
+
+type addrKeyWrapper struct {
+	addr.Address
+}
+
+func (kw addrKeyWrapper) Key() string {
+	return string(kw.Bytes())
+}
+
+func getStoragePower(rt vmr.Runtime, root cid.Cid, a addr.Address) (abi.StoragePower, bool) {
+	hm := adt.NewMap(vmr.AsStore(rt), root)
+
+	var out abi.StoragePower
+	err := hm.Get(asKeyer(a), &out)
+	if err == hamt.ErrNotFound {
+		return abi.NewStoragePower(0), false
+	}
+	requireNoStateErr(rt, err, "failed to get storage power for address %v from claimed power HAMT", a)
```

Comment:
After discussion with @whyrusleeping I think he'd rather we propagate the errors here up to the top level.

Reply:
Sounds good, will do that in a follow on as that on its own is a fairly large change.